### DISCUSSION TITLE
自定义路由模式无法代理局域网 IP

### DISCUSSION THREAD

**Discussion by MkQtS** [immortalwrt/homeproxy#101] [Q&A] [answered]
我用 [zju-connect](https://github.com/Mythologyli/zju-connect) 在路由器上搭建了一个 Socks 节点，可以连接到校园局域网。CC98 是一个内网论坛，`www.cc98.org` 的 IP 是 `10.10.98.98`

homeproxy 使用自定义路由，相关配置：

```
root@OpenWrt:~# uci show homeproxy.ZJUroute
homeproxy.ZJUroute=routing_rule
homeproxy.ZJUroute.label='ZJUroute'
homeproxy.ZJUroute.enabled='1'
homeproxy.ZJUroute.mode='default'
homeproxy.ZJUroute.source_ip_is_private='0'
homeproxy.ZJUroute.ip_is_private='1'
homeproxy.ZJUroute.rule_set='ZJU_list'
homeproxy.ZJUroute.rule_set_ipcidr_match_source='0'
homeproxy.ZJUroute.outbound='sbRVPN'
homeproxy.ZJUroute.ip_cidr='10.0.0.0/8'
```

https://github.com/immortalwrt/homeproxy/blob/4fc26d5cd5583cce3581253dffca2e127c72846a/root/etc/homeproxy/scripts/firewall_post.ut#L94-L119

这段代码似乎阻止了代理 私有 IP，于是我无法访问 CC98

删去`/etc/homeproxy/scripts/firewall_post.ut`里的`10.0.0.0/8,`后可以访问

或许 homeproxy 应该在发现设置了 `homeproxy.ZJUroute.ip_is_private='1'` 后允许代理相关私有 IP？

**Comment by 1715173329** [comment, marked as answer]
绕过不是阻止。你可以自行把相关 ip 加入到 访问控制 里面的 代理 IP 地址

**Reply by MkQtS** [reply]
好吧，没有想到

![image](https://github.com/immortalwrt/homeproxy/assets/81752398/98486ccf-9a96-447a-b3d3-586788ab84b2)

路由规则这里可以加个提示，比如“默认不代理私有 IP，需要将相关 IP 添加到 访问控制 - 代理 IP”

**Reply by tanst** [reply]
> 需要将相关 IP 添加到 访问控制 - 代理 IP”

这是加到 `WAN IP 策略` 里面？

**Comment by bbpp222006** [comment]
测试了一下，若需要分流局域网IP（比如内网跳板之类的需求），需要配置这些内容（以分流10.0.0.0/8为例）：
1. 访问控制-WAN IP 策略 -代理 IP 添加10.0.0.0/8
2. 路由规则内勾选（私有ip）一项，并增加10.0.0.0/8
3. 选择 路由设置-路由端口-所有端口 (若为自定义端口，比如23333，则此项需要配置)
